lib/awful/screen.lua

```diff
@@ -234,6 +235,63 @@ function screen.object.set_padding(self, padding)
     end
 end
 
+--- A list of outputs for this screen with their size in mm.
+--
+-- This information is only available when RANDR is used to manage the screens.
+--
+-- Please note that the table content may vary.
+--
+-- An easy way to check if a screen is the laptop screen is usually:
+--
+--    if s.outputs["LVDS-1"] then
+--        -- do something
+--    end
+--
+-- **Signal:**
+--
+--  * *property::outputs*
+--
+-- **Immutable:** true
+-- @property outputs
+-- @param table
+-- @tfield table table.name A table with the screen name as key (like `eDP1` on a laptop)
+-- @tfield integer table.mm_width The screen physical width
+-- @tfield integer table.mm_height The screen physical height
+-- @tfield integer table.name The screen physical height
+-- @tfield integer table.viewport_id The identifier of the viewport this output
+--  corresponds to.
```

Comment:
Why just an identifier instead of providing the viewport directly? (I guess whatever a viewport is, it has some kind of `.id` property, so nothing is lost here)

Reply:
To avoid using "miss_handlers" or "valid" because they are convoluted anti patterns and I prefer not to add more of them. It also makes improving the API harder because the code to handle the legacy is horrifying.

I added a comment in the code to reflect that design choice.